torch/_tensor_str.py

```diff
@@ -115,7 +115,7 @@ def printoptions(**kwargs):
 
 
 def tensor_totype(t):
-    dtype = torch.float if t.is_mps else torch.double
+    dtype = torch.float if (t.is_mps or t.is_xpu) else torch.double
```

Comment:
You don't have double precision dtypes?

Reply:
Let me clarify that, in general, enterprise-grade GPU have double precision types, while consumer-grade GPU do not.

Reply:
If you will change precision here, then some calculations might start overflow and final printout might confuse users. Also, as you just commented, some of XPU devices has fp64 support, some - do not. Thus, checking for `is_xpu` is not a fine grained check.

I would go with double as much as possible. For XPU we can have a finer grain check using `torch.xpu.get_device_properties().has_fp64`.

Reply:
It's better if we can add the check of has_fp64 here. @guangyey  pls. help confirm.

BTW, it should be safe if not convert to double, as most of low precision float data types have been converted to float already before print?
https://github.com/pytorch/pytorch/blob/168e41009bbd664c805a6078ed20d44ffe101335/torch/_tensor_str.py#L332

Reply:
> BTW, it should be safe if not convert to double, as most of low precision float data types have been converted to float already before print?

because we do reduction to compute extremum value, we have been on the safe side of using double to reduce the change of floating point artifacts.